Clarify authentication requriements (#239)
oauth-wg/oauth-transaction-tokens: draft-ietf-oauth-transaction-tokens.md

The unsigned JSON object MAY contain other fields, and the Txn-Token Service MAY consider them when generating the Txn-Token.

## Txn-Token Request Processing

> When the Transaction Token Service receives a Txn-Token Request it MUST authenticate the workload and determine if that workload is authorized to obtain the Txn-Tokens with the requested values. The authorization policy for determining such issuance is out of scope for this specification.

Reviewer: "the Txn-Tokens" sounds not quite right to my eyes 
```suggestion
When the Transaction Token Service receives a Txn-Token Request it MUST authenticate the workload and determine if that workload is authorized to obtain Txn-Tokens with the requested values. The authorization policy for determining such issuance is out of scope for this specification.
```

A successful response by the Txn-Token Service to a Replacement Txn-Token Request is a Txn-Token Response as described in {{txn-token-response}}

## Mutual Authentication of the Txn-Token Request
A Transaction Token Service and requesting workload MUST authenticate each other (mutual authentication). Workloads MUST authenticate the Transaction Token Service to ensure that they do not disclose sensitive information, such as OAuth access tokens, to a rogue Transaction Token Service. A Transaction Token Service MUST authenticate a workload to ensure the workload is authorized to request a transaction token.

> Workloads SHOULD use the `https` scheme to secure the communication channel and authenticate the Transaction Token Service. When using `https`, TLS certificates MUST be checked according to {{RFC9110}}. At the time of this writing, TLS version 1.3 {{RFC8446}} is the most recent version

Reviewer: I don't think RFC 9110 is right for checking tls certificates. Also need a period. 
```suggestion
Workloads SHOULD use the https scheme to secure the communication channel and authenticate the Transaction Token Service. When using https, TLS certificates MUST be validated according to {{RFC6125}}. At the time of this writing, TLS version 1.3 {{RFC8446}} is the most recent version.
```

Author: @bc-pi - I was trying to be consistent with OAuth 2.1: https://www.ietf.org/archive/id/draft-ietf-oauth-v2-1-13.html#name-communication-security

"When using https, TLS certificates MUST be checked according to [[RFC9110](https://www.ietf.org/archive/id/draft-ietf-oauth-v2-1-13.html#RFC9110)]."

RFC 9110 does refer to RFC6125 for cert validation...

Reviewer: I would have had the same feedback for @aaronpk and his OAuth 2.1 coauthors (but really only Aaron b/c AFAICT he's the only one really contributing to 2.1) but admittedly I did miss the RFC6125 cert validation stuff in the nearly 200 pages of RFC9100. Apologies there. 

Perhaps just a bit more specific reference (I think/hope this syntax is correct)? 

```suggestion
Workloads SHOULD use the `https` scheme to secure the communication channel and authenticate the Transaction Token Service. When using `https`, TLS certificates MUST be checked according to {{Section 4.3.4 or RFC9110}}. At the time of this writing, TLS version 1.3 {{RFC8446}} is the most recent version.
```

Ignoring that I just noticed again that RFC6125 has been obsoleted by RFC 9525. But maybe pointing at 9110 can enable just continuing to ignore that.

A successful response by the Txn-Token Service to a Replacement Txn-Token Request is a Txn-Token Response as described in {{txn-token-response}}

## Mutual Authentication of the Txn-Token Request
A Transaction Token Service and requesting workload MUST authenticate each other (mutual authentication). Workloads MUST authenticate the Transaction Token Service to ensure that they do not disclose sensitive information, such as OAuth access tokens, to a rogue Transaction Token Service. A Transaction Token Service MUST authenticate a workload to ensure the workload is authorized to request a transaction token.

Workloads SHOULD use the `https` scheme to secure the communication channel and authenticate the Transaction Token Service. When using `https`, TLS certificates MUST be checked according to {{RFC9110}}. At the time of this writing, TLS version 1.3 {{RFC8446}} is the most recent version

> Workloads SHOULD authenticate to a Transaction Token Server using asymmetric (public-key based) methods or signed JWTs in accordance with {{RFC7521}} and {{RFC7523}}.

Reviewer: ```suggestion
Workloads SHOULD authenticate to a Transaction Token Server using asymmetric (public-key based) methods or signed client authentication JWTs in accordance with {{RFC7523}}.
```

Author: Of course - 7521 does not talk about JWTs...

> Examples of public-key based authentication include those defined in OAuth 2.0 Mutual-TLS Client Authentication and Certificate-Bound Access Tokens {{RFC8705}} and WIMSE Workload-to-Workload Authentication {{WIMSE}}.

Reviewer: related to previous comment on `WIMSE: ...` above 
```suggestion
Examples of public-key based authentication include those defined in OAuth 2.0 Mutual-TLS Client Authentication and Certificate-Bound Access Tokens {{RFC8705}} and WIMSE Workload-to-Workload Authentication {{?I-D.ietf-wimse-s2s-protocol}}.
```

A successful response by the Txn-Token Service to a Replacement Txn-Token Request is a Txn-Token Response as described in {{txn-token-response}}

## Mutual Authentication of the Txn-Token Request
A Transaction Token Service and requesting workload MUST authenticate each other (mutual authentication).

> Workloads authenticate the Transaction Token Service to ensure that it does not disclose sensitive information, such as OAuth access tokens, to a rogue Transaction Token Service. A Transaction Token Service authenticates a workload to ensure the workload is authorized to request a transaction token.

Author: Thanks @yaronf

## Subject Token Types {#sec-sub-token-types}
A service requesting a Txn-Token SHOULD provide an incoming token if it has one that it used itself to authorize a caller, and if it directly correlates with the downstream call chain it needs the Txn-Token for. In the absence of an appropriate incoming token, the requesting service MAY use a self-signed JWT, an unsigned JSON object or any other format to represent the details of the requester to the Txn-Token service.

## Use of 'actor_token'

> If using the `actor_token` and `actor_token_type` parameters of the OAuth 2.0 Token Exchange specification {{RFC8693}}, both parameters MUST be present in the request. The `actor_token` MUST authenticate the identity of the requesting workload.

Reviewer: does it have to? wouldn't this preclude the `actor_token` being used in conjunction with some other form of authentication? 

```suggestion
If using the `actor_token` and `actor_token_type` parameters of the OAuth 2.0 Token Exchange specification {{RFC8693}}, both parameters MUST be present in the request. The `actor_token` can authenticate the identity of the requesting workload.
```
